src/services/completions.ts

```diff
@@ -2197,4 +2198,28 @@ namespace ts.Completions {
     function hasIndexSignature(type: Type): boolean {
         return !!type.getStringIndexType() || !!type.getNumberIndexType();
     }
+
+    function isInvalidTrigger(sourceFile: SourceFile, triggerCharacter: string | undefined, contextToken: Node, position: number): boolean {
+        switch (triggerCharacter) {
+            case '"':
+            case "'":
+            case "`":
+                switch (contextToken.kind) {
+                    case SyntaxKind.StringLiteral:
+                    case SyntaxKind.NoSubstitutionTemplateLiteral:
+                    case SyntaxKind.TemplateExpression:
+                    case SyntaxKind.TaggedTemplateExpression:
+                        // Don't automatically bring up completions at closing quote
+                        return position === contextToken.parent.end;

```

Comment:
not to be pedantic, but you might want to check the first char in the string to match the trigger char to allow for things like:
```ts
var x: "`a`" | "`b`" = "`
```